isthmus/src/main/java/io/substrait/isthmus/TypeConverter.java

```diff
@@ -100,43 +129,53 @@ private static Type convert(RelDataType type, List<String> names) {
       case BINARY -> creator.fixedBinary(type.getPrecision());
       case MAP -> {
         MapSqlType map = (MapSqlType) type;
-        yield creator.map(convert(map.getKeyType(), names), convert(map.getValueType(), names));
+        yield creator.map(
+            toSubstrait(map.getKeyType(), names), toSubstrait(map.getValueType(), names));
       }
       case ROW -> {
         var children = new ArrayList<Type>();
         for (var field : type.getFieldList()) {
           names.add(field.getName());
-          children.add(convert(field.getType(), names));
+          children.add(toSubstrait(field.getType(), names));
         }
         yield creator.struct(children);
       }
-      case ARRAY -> creator.list(convert(type.getComponentType(), names));
+      case ARRAY -> creator.list(toSubstrait(type.getComponentType(), names));
       default -> throw new UnsupportedOperationException(
           String.format("Unable to convert the type " + type.toString()));
     };
   }
 
-  public static RelDataType convert(RelDataTypeFactory type, TypeExpression typeExpression) {
-    return convert(type, typeExpression, null);
+  public RelDataType toCalcite(
+      RelDataTypeFactory relDataTypeFactory, TypeExpression typeExpression) {
+    return toCalcite(relDataTypeFactory, typeExpression, null);
```

Comment:
This takes a `RelDataTypeFactory`, rather than bundling it in the `TypeConverter` itself, because not everywhere where you would want to build a `TypeConverter` has easy access to one of these factories, but everywhere you want to call `toCalcite` does.